api/v1beta1/module_webhook.go

```diff
@@ -0,0 +1,101 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	"regexp"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var modulelog = logf.Log.WithName("module-resource")
+
+func (m *Module) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(m).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-kmm-sigs-x-k8s-io-v1beta1-module,mutating=false,failurePolicy=fail,sideEffects=None,groups=kmm.sigs.x-k8s.io,resources=modules,verbs=create;update,versions=v1beta1,name=vmodule.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Module{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (m *Module) ValidateCreate() error {
+	modulelog.Info("validate create", "name", m.Name)
+
+	if err := m.validateKernelMappingRegex(); err != nil {
+		return fmt.Errorf("failed to validate the kernel mapping regexps: %v", err)
+	}
+
+	return m.validateModprobe()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (m *Module) ValidateUpdate(_ runtime.Object) error {
+	modulelog.Info("validate update", "name", m.Name)
+
+	if err := m.validateKernelMappingRegex(); err != nil {
+		return fmt.Errorf("failed to validate the kernel mapping regexps: %v", err)
+	}
+
+	return m.validateModprobe()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (m *Module) ValidateDelete() error {
+	return nil
+}
+
+func (m *Module) validateKernelMappingRegex() error {
+	container := m.Spec.ModuleLoader.Container
+
+	for idx, km := range container.KernelMappings {
+		if _, err := regexp.Compile(km.Regexp); err != nil {
+			return fmt.Errorf("invalid regexp at index %d: %v", idx, err)
+		}
+
+		if container.ContainerImage == "" && km.ContainerImage == "" {
+			return fmt.Errorf("missing spec.moduleLoader.container.kernelMappings[%d].containerImage", idx)
+		}
+	}
+
+	return nil
+}
+
+func (m *Module) validateModprobe() error {
+	modprobe := m.Spec.ModuleLoader.Container.Modprobe
+
+	if modprobe.ModuleName != "" {
+		return nil
+	}
+
+	if modprobe.RawArgs == nil {
+		return errors.New("missing spec.moduleLoader.container.modprobe.moduleName")
+	}
+
+	if len(modprobe.RawArgs.Load) == 0 || len(modprobe.RawArgs.Unload) == 0 {
+		return errors.New("when using spec.moduleLoader.container.modprobe.rawArgs, both load and unload fields should be specified")
+	}
```

Comment:
I think that `moduleName` and `rawArgs` are mutually exclusive so I would go with:
* If both are not set - return an error
* If both are set - return an error

At this point we can assume that exactly one of them is set, then:
* if moduleName is set return nil

At this point we can assume that `rawArgs` is set
* make sure that both `load` and `unload` are set.